src/Sulu/Bundle/RouteBundle/PageTree/PageTreeTrait.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\PageTree;
+
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentInspector;
+use Sulu\Bundle\RouteBundle\Content\Type\PageTreeRouteContentType;
+use Sulu\Bundle\RouteBundle\Document\Subscriber\RoutableSubscriber;
+use Sulu\Component\Content\Metadata\PropertyMetadata;
+use Sulu\Component\Content\Metadata\StructureMetadata;
+use Sulu\Component\Route\Document\Behavior\RoutableBehavior;
+
+/**
+ * Encapsulates function to extract parent page from structure-metadata.
+ */
+trait PageTreeTrait
```

Comment:
Where is that actually used? I can't find another occurrence of this text...

Reply:
It's only used by the article bundle so far, but it belongs to the RouteBundle in my opinion

Reply:
ArticleIndexer and WebspaceSubscriber